tests/conftest.py

```
@@ -51,6 +52,11 @@ def pytest_configure(config):
     compares against the mocked "xpu" and would return False.  This is the known
     trade-off: routing logic is bypassed and must be tested separately (Option 1).
     """
+    # Give each torchrun rank its own basetemp to avoid cleanup race conditions
+    rank = os.environ.get("LOCAL_RANK")
+    if rank is not None and config.option.basetemp is None:
+        config.option.basetemp = f"/tmp/pytest-torchrun-rank{rank}"
```

Ruff (0.15.18)

[error] Probable insecure usage of temporary file or directory: "/tmp/pytest-torchrun-rank"

(S108)